cmd/server/handler/warehouses_test.go

```diff
@@ -0,0 +1,299 @@
+package controllers_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/marcoglnd/mercado-fresco-packmain/cmd/server/routes"
+	"github.com/marcoglnd/mercado-fresco-packmain/internal/warehouses"
+	"github.com/stretchr/testify/assert"
+)
+
+func getPathUrl(url string) string {
+	PATH := "/api/v1"
+	return fmt.Sprintf("%s%s", PATH, url)
+}
+
+func createServer() *gin.Engine {
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	routerGroup := router.Group(getPathUrl(""))
+	routes.AddRoutes(routerGroup)
+
+	return router
+}
+
+func createRequestTest(
+	method string,
+	url string,
+	body string,
+) (*http.Request, *httptest.ResponseRecorder) {
+	req := httptest.NewRequest(method, url, bytes.NewBuffer([]byte(body)))
+	req.Header.Add("Content-Type", "application/json")
+
+	return req, httptest.NewRecorder()
+}
```

Comment:
About this `createRequestTest` function, looking that all tests needed it. What do you think about moving it into an `index_test.go` file